### PR TITLE
Add smoke test for preagg metrics with sampling and overrides enabled

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java [hunk 1]
```diff
@@ -444,6 +448,8 @@ public static class LegacyRequestIdPropagation {
   public static class InternalConfiguration {
     // This is used for collecting internal stats
     public Statsbeat statsbeat = new Statsbeat();
+    public PreAggregatedMetricsConfiguration preAggregatedMetric =
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:451]:
```suggestion
    public PreAggregatedMetricsConfiguration preaggregatedStandardMetrics =
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 2]
```diff
@@ -531,6 +531,15 @@ static void overlayFromEnv(Configuration config, Path baseDir) throws IOExceptio
...
     overlayInstrumentationEnabledEnvVars(config);
+
+    // disable pre-aggregated metric if sampling rate is not 100 and
+    // roleName/connectionString/instrumentationKey overrides are used
+    if ((config.preview.roleNameOverrides.size() != 0
+            || config.preview.connectionStringOverrides.size() != 0
+            || config.preview.instrumentationKeyOverrides.size() != 0)
+        && config.sampling.percentage != 100) {
+      config.internal.preAggregatedMetric.disabled = true;
+    }
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:542]: I think this is solved by #2523, let's remove this (but keep the kill switch and the disabled smoke test), and then let's merge and I'll pull the smoke test into #2523 to test there

**Review by trask — APPROVED** [review]

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java (hunk 2 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: done

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java (hunk 1 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java]: Renamed.

File: smoke-tests/apps/PreAggMetricsWithRoleNameOverridesAndSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PreAggMetricsWithRoleNameOverridesAndSamplingTest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        testing.mockedIngestion.getItemsEnvelopeDataType("RequestData");
+    Thread.sleep(SECONDS.toMillis(10));
+
+    List<Envelope> metricsEnvelops = testing.mockedIngestion.getItemsEnvelopeDataType("MetricData");
+    // TODO need to fix pre agg metrics for non-sampled spans (PropagatedMetric) with role name
+    // overrides
+    List<Envelope> clientRoleNameOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> clientRoleNameNotOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> serverRoleNameOverriddenEnvelops = new ArrayList<>();
+    List<Envelope> serverRoleNameNotOverriddenEnvelops = new ArrayList<>();
```
Comment (heyams) [inline smoke-tests/apps/PreAggMetricsWithRoleNameOverridesAndSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PreAggMetricsWithRoleNameOverridesAndSamplingTest.java:71]: need to update this when testing your pr @trask. here is 50% rolenames are updated.

Comment (heyams) [reply smoke-tests/apps/PreAggMetricsWithRoleNameOverridesAndSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/PreAggMetricsWithRoleNameOverridesAndSamplingTest.java:71]: you can merge yours first and then i can update here.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java [hunk 4]
```diff
@@ -446,6 +450,8 @@ public static class LegacyRequestIdPropagation {
   public static class InternalConfiguration {
     // This is used for collecting internal stats
     public Statsbeat statsbeat = new Statsbeat();
+    public PreAggregatedStandardMetricsConfiguration preAggregatedStandardMetric =
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:453]: (lowercase "a" and trailing "s")
```suggestion
    public PreAggregatedStandardMetricsConfiguration preaggregatedStandardMetrics =
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java [hunk 5]
```diff
@@ -272,6 +272,10 @@ public static class Statsbeat {
     public long longIntervalSeconds = DAYS.toSeconds(1); // default to daily
   }
 
+  public static class PreAggregatedStandardMetricsConfiguration {
+    public boolean disabled = false; // pre-aggregated metrics are on by default
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java:276]: we generally go with `enabled` flag, even if it's enabled (`true`) by default
```suggestion
    public boolean enabled = true; // pre-aggregated metrics are on by default
```

**Review by trask — APPROVED** [review]

**Pull request merged**